client/rpmtrans.c

```diff
@@ -44,6 +44,7 @@ TDNFRpmExecTransaction(
                   1,
                   sizeof(TDNF_CACHED_RPM_LIST),
                   (void**)&ts.pCachedRpmsArray);
+    BAIL_ON_TDNF_ERROR(dwError);
```

Comment:
Missing error check

Reply:
yes. please remove if. bail (that you added) is enough.

Reply:
This is done, please have a look once you're free, thanks for the quick response and reviews :) It's almost 1.30AM here, feeling sleepy, if any modifications needed, I'll do it tomorrow.